### ISSUE TITLE
extract_from_roi 

### ISSUE THREAD

**Issue by YasaswiniSireddy** [deepdoctection/deepdoctection#367]
I'm are using a custom Layout Parser model, which is registered and has text, title, table. ... as categories.
I am trying to use pdfplumber detector and textextractionservice. 
Code :
 from deepdoctection.extern.pdftext import PdfPlumberTextDetector
from deepdoctection.pipe.text import TextExtractionService
import deepdoctection as dd

dd.ModelCatalog.register("layoutparser/publaynet/model_final.pth", dd.ModelProfile(
    name="layoutparser/publaynet/model_final.pth",
    description="Layout analysis",
    config="layoutparser/publaynet/config.yml",
    tp_model=False,
    size=[],
    categories={

        1: dd.LayoutType.TEXT,
        2: dd.LayoutType.TITLE,
        3: dd.LayoutType.LIST,
        5: dd.LayoutType.TABLE,

    },
    model_wrapper="D2FrcnnDetector",
))

# Initialize components
path_weights = dd.ModelCatalog.get_full_path_weights("layoutparser/publaynet/model_final.pth")
path_configs = dd.ModelCatalog.get_full_path_configs("layoutparser/publaynet/model_final.pth")
categories = dd.ModelCatalog.get_profile("layoutparser/publaynet/model_final.pth").categories

d2_detector = dd.D2FrcnnDetector(path_configs, path_weights, categories, device="cpu")
image_layout = dd.ImageLayoutService(d2_detector)
pdf_plumber = PdfPlumberTextDetector()
text_extract = TextExtractionService(pdf_plumber,extract_from_roi=dd.LayoutType.TITLE)

pipe = dd.DoctectionPipe([text_extract])

df = pipe.analyze(path="/data/foo.pdf")
for dp in df:
    print(dp.text)

![code](https://github.com/user-attachments/assets/b208b1cd-de63-488d-97f7-b41b050832b3)

When i try to send some categories using 'roi_ ....' parameter',
 i get this error : 
 Traceback (most recent call last):
  File "/home/yasaswinisireddy/anaconda3/envs/docAi/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/yasaswinisireddy/anaconda3/envs/docAi/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/yasaswinisireddy/Python Ml/deepdoctection/Examples/test.py", line 33, in <module>
    text_extract = TextExtractionService(pdf_plumber,extract_from_roi=dd.LayoutType.TITLE)
  File "/home/yasaswinisireddy/Python Ml/deepdoctection/deepdoctection/pipe/text.py", line 98, in __init__
    raise TypeError(
TypeError: Predicting from a cropped image requires to pass an ObjectDetector or TextRecognizer. Got <class 'deepdoctection.extern.pdftext.PdfPlumberTextDetector'>

Question : what exactly extract_from_roi is doing? how can i use it to pass the category_names like LayoutType.TEXT?

**Comment by JaMe76** [comment]
Suppose you have an image (not a native PDF but something like a .PNG, .JPG) and you have a layout detector partitioning your document into text blocks, title, tables, figures. With an OCR detector (e.g. Tesseract, DocTR) you could try to run OCR only on the layouts segments. If you want to get the text only from figures you need to make sure to create the sub-images by cropping all figures off the full image and then pass only the sub-images to the OCR engine. 

```python
d2_detector = D2FrcnnDetector(path_configs, path_weights, categories, device="cpu")
image_layout = ImageLayoutService(d2_detector, to_image=True, crop_image=True)
ocr_engine = TesseractOcrDetector()
text_extract = TextExtractionService(ocr_engine,extract_from_roi=[LayoutType.FIGURE])
```

The arguments  `to_image=True, crop_image=True` in  `ImageLayoutService` will ensure to build sub-images from detected layout segments. `extract_from_roi=[LayoutType.FIGURE]` will ensure to pass only figure elements to the OCR engine. Every other layout sections will not be passed to the OCR engine.

The issue you run into comes from the fact that you use `PdfPlumberTextDetector` which is not an OCR engine. That means it does not run on an numpy array but on a bytes stream encoded by the native PDF. This is the reason why you get the error message that `extract_from_roi` and `PdfPlumberTextDetector` are not compatible. You cannot run `PdfPlumberTextDetector` on a sub-image. You can only run `PdfPlumberTextDetector` on the full PDF page and then match the extracted text to all layout segments. 

In summary: Either change the text extractor ans use an OCR engine or remove `extract_from_roi`.

**Comment by YasaswiniSireddy** [comment]
Got it so we can not use extract_from_roi for PdfPlumberTextDetector.
Thank You Very Much for the quick response.
